File: ruby/arrays/hashes/merging_hashes.rb
#!/usr/bin/ruby

things = {
  'fruit' => 'grape',
  'utensil' => 'spoon',
  'garment' => 'scarf',
}
items = {
    2 => "Hello, there!",
    'color' => 'orange',
    3.14 => 'PI',
    'plant' => 'grape',
    'utensil' => 'spoon',
}

puts("things: #{things}")
puts("items: #{items}\n\n")

elements = things.merge(items);

puts("elements = things.merge(items)")
puts("elements: #{elements}\n\n")

elements = items.merge(things) { |key, items_element, things_element| key + ':' + items_element + '&' + things_element }

puts("elements = items.merge(things) { |key, items_element, things_element| key + ':' + items_element + '&' + things_element }")
puts("elements: #{elements}\n\n")

things.merge!(items)

puts("things.merge!(items)")
puts("things: #{things}\n\n")

things = {
  'fruit' => 'grape',
  'utensil' => 'spoon',
  'garment' => 'scarf',
}

items.merge!(things) { |key, items_element, things_element| key + ':' + items_element + '&' + things_element }

puts("items.merge!(things) { |key, items_element, things_element| key + ':' + items_element + '&' + things_element }")
puts("items: #{items}\n\n")

File: perl/arrays/hashes/merging_hashes.pl
#
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
#

use strict;
use warnings;

my %things = (
  'fruit' => 'grape',
  'utensil' => 'spoon',
  'garment' => 'scarf',
);
my %items = (
    2 => "Hello, there!",
    'color' => 'orange',
    3.14 => 'PI',
    'plant' => 'grape',
    'utensil' => 'cup',
);

print("\%things: @{[%things]}\n");
print("\%items: @{[%items]}\n\n");

my %elements = (%things, %items);

print("\%elements = (\%things, \%items);\n");
print("\%elements: @{[%elements]}\n\n");
